lib/tasks_private/openstack.rake

```diff
@@ -32,54 +28,77 @@ namespace :vcr do
-      sg_id = sg.data.dig(:body, "security_group", "id")
+      cn = network.create_network(:name => "rake-network").body["network"]
+      cn_id = cn["id"]
+      tenant_id = cn["tenant_id"]
+      network.subnets.create(:network_id => cn_id, :cidr => "10.0.0.0/21", :ip_version => 4)
+      network.create_router("rake-router")
+      network.create_security_group(:name => "rake-sg", :description => "test description", :tenant_id => tenant_id)
 
       image = compute.images.first
       flavor = compute.flavors.get(1)
-      vm = compute.servers.create(:name => 'test-server', :flavor_ref => flavor.id, :image_ref => image.id, :nics => [{:net_id => cn_id}])
-      compute.create_key_pair("test-key")
-      vol = compute.create_volume("test-vol", "test description", 1)
-      vol_id = vol.data.dig(:body, "volume", "id")
-      agg = compute.create_aggregate("test-aggregate")
-      agg_id = agg.data.dig(:body, "aggregate", "id")
+      compute.servers.create(:name => 'rake-server', :flavor_ref => flavor.id, :image_ref => image.id, :nics => [{:net_id => cn_id}])
+      compute.create_key_pair("rake-key")
+      compute.create_volume("rake-vol", "test description", 1)
+      compute.create_aggregate("rake-aggregate")
     rescue => err
       puts err
     end
 
-    puts "Generating VCRs..."
-    `bundle exec rspec ./spec/models/manageiq/providers/openstack/cloud_manager/refresher_spec.rb`
+    desc 'Cleanup resources from environment'
+    task :cleanup => :environment do
+      vm = compute.list_servers.body["servers"].find { |vms| vms["name"] == "rake-server" }
+      vol = compute.list_volumes.body["volumes"].find { |vols| vols["displayName"] == "rake-vol" }
+      agg = compute.list_aggregates.body["aggregates"].find { |aggs| aggs["name"] == "rake-aggregate" }
+      cn = network.list_networks.body["networks"].find { |cns| cns["name"] == "rake-network" }
+      cs = network.list_subnets.body["subnets"].find { |subnets| subnets["network_id"] == cn["id"] }
+      router = network.list_routers.body["routers"].find { |routers| routers["name"] == "rake-router" }
+      sg = network.list_security_groups.body["security_groups"].find { |sgs| sgs["name"] == "rake-sg" }
 
-    puts "Cleaning up resources..."
-    compute.delete_server(vm.id)
-    compute.delete_key_pair("test-key")
-    compute.delete_volume(vol_id)
-    compute.delete_aggregate(agg_id)
+      puts "Cleaning up resources..."
+      compute.delete_server(vm["id"]) unless vm.nil?
+      compute.delete_key_pair("rake-key")
+      compute.delete_volume(vol["id"]) unless vol.nil?
+      compute.delete_aggregate(agg["id"]) unless agg.nil?
 
-    10.times do
-      sleep(10)
-    end
+      10.times do
+        break if vm.nil?
 
-    network.delete_subnet(cs.id)
+        compute.get_server_details(vm["id"])
+        sleep(10)
+      rescue Fog::Compute::OpenStack::NotFound
+        break
+      end
```

Comment:
I think we could use a method to share this logic, similar to the `with_retry` we have in `VPC`